### ISSUE TITLE
Why use relative path to assets?

### ISSUE THREAD

**Issue by gurupras** [antfu/vitesse-webext#35]
I'm guessing there's a good reason as to why the following code is present:

**vite.config.js**
```ts
export const sharedConfig: UserConfig = {
    ...
    // rewrite assets to use relative path
    {
      name: 'assets-rewrite',
      enforce: 'post',
      apply: 'build',
      transformIndexHtml (html) {
        return html.replace(/"\/assets\//g, '"../assets/')
      }
    }
}
```

However, this is causing issues with my extension which is trying to create devtools panels. I'm pretty sure this is due to the structure of my panel, which is under `src/devtools/panels/index.html`, which ends up referencing `dist/devtools/assets` instead of `dist/assets`.

Could you please share some information as to why this decision was made? Perhaps throw this rationale in the code via comments so others are made aware of it as well?

**Comment by tmkx** [comment]
It's only replacing `"/assets/` to `"../assets/`, for example: `<script type="module" crossorigin src="/assets/devtools.ae186893.js"></script>` will be replaced to `<script type="module" crossorigin src="../assets/devtools.ae186893.js"></script>`, `dist/devtools/assets` won't be replaced.

Can you provide a reproduce repo?

it works for me:
![image](https://user-images.githubusercontent.com/13805071/134312596-eec7de5c-7421-44d5-98f8-13c2f23ce7bf.png)

**Comment by gurupras** [comment]
Sorry, this is my first time working with a lot of the new tools in this template (TS, Vite, etc).

I have an index.html in the same structure as your image and that one works fine. However, I also have a deeper one:
```
src
├── background
├── contentScripts
├── components
└── devtools
    ├── index.html
    └── panels
        └── index.html
```
This is failing for the `src/devtools/panels/index.html` file as shown below:
![webext-panels](https://user-images.githubusercontent.com/2933207/134367381-7551e344-67f2-4cd7-9b17-fbb164eafc98.png)

This structure fails during development (`npm run dev`) with the following error:
> Refused to load the script 'http://localhost:3303/devtools/main.ts' because it violates the following Content Security Policy directive: "script-src 'self' blob: filesystem:". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.

Everything works again if I change the rule in `vite.config.js` to be:
```ts
// rewrite assets to use relative path
{
  name: 'assets-rewrite',
  enforce: 'post',
  apply: 'build',
  transformIndexHtml (html) {
    // return html.replace(/"\/assets\//g, '"../assets/') // <-- Does not work
    return html.replace(/"\/assets\//g, isDev ? '"../assets/' : '"/dist/assets/')  // Works
  }
}
```

**Comment by tmkx** [comment]
I see

**Comment by gurupras** [comment]
I'm seeing an issue with this even after #36.

In production builds of my extension, I'm seeing the following:
![image](https://user-images.githubusercontent.com/2933207/134759290-1db490e8-d502-40c9-a1de-c1826714b3eb.png)

**Comment by gurupras** [comment]
Another quick update:
Something in the bundling process is messing up the import paths. For example, in one of my generated JS files, I see the following snippet:
```js
const{onMessage:onMessage,sendMessage:sendMessage}=await(baseModule=()=>import("./vendor.d3042df0.js").then(
(function(n){return n.N})),deps=["assets/vendor.d3042df0.js","assets/vendor.1f19cb17.css"]
```
This ends up making network requests for:
> chrome-extension://ljdojmnoccfgmmgdjgbhbkakbanmbpbg/assets/vendor.d3042df0.js
> chrome-extension://ljdojmnoccfgmmgdjgbhbkakbanmbpbg/assets/vendor.1f19cb17.css

Both of which fail due to the missing `/dist/` prefix.

I don't know how to fix the issue, but I'm happy to help investigate/debug in any way I can. Let me know how I can help.

**Comment by tmkx** [comment]
@gurupras Could you please provide a reproduce repo?

**Comment by gurupras** [comment]
Could you try this repo? https://github.com/gurupras/message-service-extension

**Comment by gurupras** [comment]
Another minor update:
This doesn't have anything to do with my repo structure. I even moved my `src/devtools/panels/*` to be more like the other files and sit at `src/panels/*` and this still produces the bad chunk just like above.

**Comment by tmkx** [comment]
I come to the same conclusion that it's not caused by the deeper entry.

It's more likely being caused by the base of assets:
![image](https://user-images.githubusercontent.com/13805071/134773215-64843276-b573-4f99-92b6-fdd9092ee9be.png)


@gurupras  could you try this way of resolution? (I couldn't run your code properly in my computer, so I don't verify it 😅)
```diff
 export default defineConfig(({ command }) => ({
   ...sharedConfig,
-  base: command === 'serve' ? `http://localhost:${port}/` : undefined,
+  base: command === 'serve' ? `http://localhost:${port}/` : '/dist',
   server: {
```

**Comment by gurupras** [comment]
That works!
In fact it even solved some of the web worker issues I've been struggling with as well :cry:

**Comment by gurupras** [comment]
Actually, there's still a remaining issue with web workers in dev environment. I suspect this is also very related to some sort of relative paths.
![image](https://user-images.githubusercontent.com/2933207/134773733-648f239c-5819-46e8-8267-465bbe691831.png)

I suspect this file _should_ have been served as `localhost:3303/...` but is instead being served as `chrome-extension://<id>/panels/js`

Let me know if I should create a separate issue for this.

**Comment by tmkx** [comment]
Some workers loaded successfully?

maybe you can try this: https://vitejs.dev/guide/features.html#web-workers, or compile worker files with `tsup.config.ts`, and 
then use `new Worker(browser.runtime.getURL('/dist/path/to/workder.gobal.js'))`

**Comment by gurupras** [comment]
No, none of the workers load successfully. Some of the worker _URLs_ load, but the worker itself does not load.
Here's the content of the successful URLs:

![image](https://user-images.githubusercontent.com/2933207/134774721-51fe6a2e-b258-4d42-8287-7f75312fcfc4.png)

This references `/@fs/.../filter.worker.js?worker_file` which ends up making an XHR request for `chrome-extension://<id>/@fs/.../filter.worker.js?worker_file`

I'm probably wrong, but if I had to guess, I'd say that all dynamic imports are loading incorrectly. The `/@fs` is being treated as `chrome-extension://<id>/@fs` instead of `localhost:3303/@fs`

**Comment by tmkx** [comment]
It seems like a corner case of Vite, you can try this as a workaround: https://github.com/tmkx/vitesse-webext/commit/107b34cf88b56fc63257be0c2d57358e759ef74d
